_posts/2018-01-22-Java9-modules-Spring-Boot-2-Docker.md

```diff
@@ -0,0 +1,301 @@
+
+I played a lot with JDK 9 prereleases, betas, early access, to get hold of new things like the module system, jshell, new stream API tricks, etc. But now it's out of beta, general availability, with some updates in already. Ready to go.
+
+But it's not going to be easy. Java 9 is going to be a huge step due to its module systems and new deprecations. It will simply not run many of your existing software stacks and libraries. For example, Spring Boot 1.x is not going to work at all, and it's not going to be working at any future point either. If you wish to run Spring Boot, like many do, you need to first update Spring Boot to version 2.0. Many other libraries will break. Some have been updated, some will be updated at some point, some never will. What will help you a lot with experimentation, is Docker.
+
+## Docker
+
+It's no big secret that I love Docker. I've written a lot about it, used it extensively, and encouraged many to take it in their toolkit. It warms my heart to see that it's getting used more and more extensively these days. The reason why I love Docker for development work is because it makes experimentation much easier. For production it has much more different qualities, but they are not the topic today. If you wish to experiment with Java 9, for example, you can of course install it in your machine. But your typical installer might overwrite it as default Java, making your life much harder when you still need Java 8..7...6... for your daily work. So Docker to the rescue.
+
+This part is purely optional here, if you don't want to install Docker, or feel it's synonymous to some ugly STD, feel free to skip to next chapters. But there are many Docker images already with JDK 9 in them. And once you start one, you can run any JDK 9 tools from it. If you map your local folder to your workdir in Docker container, you can easily compile code, run maven, run your code, examine your code, test your code, etc. And of course, you can even package your .jar with JDK 9 container, and deploy that, taking full control of your OS and environment for production. You can naturally run your CI pipeline in Docker, things like Jenkins/Travis, SonarQube, run your API and E2E tests there, etc. Do not that if you're cloud fanatic like me, you can run your Docker containers easily in AWS, today also with Kubernetes support. But for your faily development needs, main thing about Docker is that you can easily try something with it, without needing to permanently install anything locally on your machine, without messing any other installations. Once you're done, you can also easily reset your Docker container.
```

Comment:
"Do not that if you're cloud fanatic like me, you can run your Docker containers easily in AWS, today also with Kubernetes support." 
Tässä alussa on varmaan jokin kirjoitusvirhe. Ei aukea mitä tarkoitat. Ehkä "Do note"?

Reply:
"But for your faily development needs," => faily => Daily